python/cymel/core/cyobjects/constraint.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+u"""
+:mayanode:`constraint`
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from ...common import *
+from ..typeregistry import nodetypes, _FIX_SLOTS
+from .cyobject import CyObject
+
+__all__ = ['Constraint']
+
+
+#------------------------------------------------------------------------------
+class Constraint(nodetypes.parentBasicNodeClass('constraint')):
+    u"""
+    :mayanode:`constraint`
+    """
+    if _FIX_SLOTS:
+        __slots__ = ('_cmd',)
+
+    def __init__(self, *args, **kwargs):
+        super(Constraint, self).__init__()
+        self._cmd = getattr(cmds, self.type())
+
+    def _getTargetList(self):
+        """Return the list of target objects.
+
+        :rtype: list[str]
+        """
+        return self._cmd(self, targetList=True, q=True)
+
+    def getTargetList(self):
+        """Return the list of target objects.
+
+        :rtype: list[Constraint]
+        """
+        return list(map(CyObject, self._getTargetList()))
+
+    def getWeightAliasList(self):
+        """
+        Returns the names of the attributes that control the weight of the target objects.
+        Aliases are returned in the same order as the targets are returned by the targetList flag
+
+        :rtype: list
+        """
+        return list(
+            map(
+                self.plug_,
+                self._cmd(self, weightAliasList=True, q=True)
+            )
+        )
```

Comment:
I noticed a discrepancy between the method name and the return value. If you need that functionality, I think a name like getWeightPlugList() would be more appropriate. Or maybe both are needed.
```suggestion
        return self._cmd(self, weightAliasList=True, q=True) or []

    def getWeightPlugList(self):
        return list(
            map(
                self.plug_,
                self._cmd(self, weightAliasList=True, q=True) or []
            )
        )
```

Reply:
updated